README_INTERFACE_TESTS.md

```diff
@@ -0,0 +1,173 @@
+    return
+```
+
+This is a complete interface test.
+
+## Validating the output state
+
+If the specification dictates something more than "the databag contents should match this schema in that event", for example:
+
+> on `ingress-relation-joined`, the requirer should write its part of the data to the relation and set its status to Active.
+
+The test would then become:
+
+```python
+from scenario import State
+from interface_tester.interface_test import interface_test_case
+
+
+@interface_test_case(
+    event='ingress-relation-joined',
+    role='requirer',
+)
+def test_data_published_on_joined(output_state: State):
+    assert output_state.status.unit.name == 'active'
+```
+
+
+## Testing a negative
+
+The second norm we have above contains a negative. We split it into two assertions to write the tests:
+
+> on `ingress-relation-changed`, if the requirer has provided 'data' and 'baz' information in its relation databags, then the provider should publish its part of the data.
+> on `ingress-relation-changed`, if the requirer has NOT provided 'data' and 'baz' information in its relation databags, then the provider should publish no data.
+
+This becomes two test cases:
+
+```python
+from scenario import State, Relation
+from interface_tester.interface_test import interface_test_case, SchemaConfig
+
+
+@interface_test_case(
+    event='ingress-relation-joined',
+    role='provider',
+    input_state=State(
+        relations=[Relation(
+            endpoint='foo',
+            interface='ingress',
+            remote_app_name='remote',  # this is our simulated requirer
+            remote_app_data={
+                'data': 'foo-bar',
+                'baz': 'qux'
+            }
+        )]
+    )
+)
+def test_data_published_on_joined_if_remote_has_sent_valid_data(output_state: State):
+    pass
+
+    
+@interface_test_case(
+    event='ingress-relation-joined',
+    role='provider',
+    schema=SchemaConfig.empty,
+    input_state=State(
+        relations=[Relation(
+            endpoint='foo',
+            interface='ingress',
+            remote_app_name='remote',
+            remote_app_data={
+                'some': 'rubbish'
+            }
+        )]
+    )
+
+)
+def test_no_data_published_on_joined_if_remote_has_not_sent_valid_data(output_state: State):
+    pass
+```
+
+The first test verifies that, if the requirer has provided correct data, then the provider will populate its side of the databag.
+The second test verifies the negative side of the norm.
+Note the usage of `SchemaConfig.empty`. That is what disables the 'default' schema validation and instructs the test runner to verify that the provider-side databags are empty instead of containing whatever they should contain according to `schema.py`.  
+
+
+## Verifying the test is discoverable
+
+If you run `interface_tester discover` in a shell (in the repo root; or check out `--help` to see how to customize that), you should see something like a list of:
```

Comment:
We're still in the `how-to` section here so I'd continue with " run X", not "if you ..."